### ISSUE TITLE
Frustrating to use

### ISSUE THREAD

**Issue by jaysnoddy** [creativecommons/ccsearch-browser-extension#254]
**Describe the bug**
This appears to have a feature I want--saving two similarly named files, one with an image and one with metadata for license and attribution, but it almost never works out in practice for me.

I can see the file I want in, say, Wikimedia Commons. But I cannot scrape the information.  Ok, I have to search for it. So...I type in all sort of metadata like the title directly from image page in wikimedia commons into your search feature, but NOTHING I enter will usually retrieve the image from the search function.  

**To Reproduce**
As just one trivial example consider the image and metadata at  https://commons.wikimedia.org/wiki/File:Sanzio_01_Plato_Aristotle.jpg 

If I try to search for "Plato and Aristotle" or a host of text variants directly taken from the page, and restrict search to Wikimedia Commons, I cannot usually find the desired file.  Somehow this search decides I probably want a statue of Confucius. or other items with a tenuous connection to the search terms.

**Additional context**
I realize the code is not likely exchangeable, but in terms of my desired functions/outputs, I wish you could, like the Zotero project, scrape data from the actual files I find on Commons.  With your tool, you cannot import/scrape the information from the source, even though I can directly see the file information.  Your tool could export the desired metadata in an easy fashion,--easier than Zotero,--but your search almost never finds the files I want to use. In contrast, you can use Zotero to easily scrape/import all sorts of useful data from a wikimedia commons page or many other pages, including rights/license, but you cannot actually use that collected data because the CSL to export does not yet seem to allow the license data it imported to actually be exported. 

** In short**
I wish you at Creative Commons could learn from Zotero's example how to scrape useful information from the actual pages with the desired content, or creative commons could work with Zotero to encourage them to actually allow the rights/licenses to be easily exported or both. They do a great job in most things, but the rights thing has been on the back burner for many years.

**Comment by makkoncept** [comment]
Hi @jaysnoddy, thanks for your feedback. I understand your problem of not getting the relevant search results but this problem is related to the [CC Catalog API](https://github.com/creativecommons/cccatalog-api), which is what this extension and [CC Search](https://search.creativecommons.org/) uses under the hood to get the search results. So, it would make sense if you open this issue on [CC Catalog API](https://github.com/creativecommons/cccatalog-api) repo as you can have a better discussion there.

**Comment by jaysnoddy** [comment]
Thanks for your helpful response!
I am, however, unsure that a better search would help with my particular workflow. Almost all the time I already know the image I want and I am at a web page with that image. Trying to re-find it again via a search would just be an unneeded step, no matter how improved that search would be.  I do not know how many potential users start with an image in mind, but I would guess many. 

What I--and others--might find useful, though, is an extension as an "attribution builder" that is somewhat similar to this extension in outputs, but assumes the user is starting out at a web page with the actual image they need.  It could facilitate choosing and downloading *one* image in web page as "filename.ext" and image metadata and attribution as filename.txt.( I am *not* talking about a bulk downloader!) It *might* make a first pass at scraping what information it could from the web page (e.g URL, Title, Organization, etc). Since it may be difficult to automate determining the license itself,  it could offer, perhaps, a checklist of CC licenses to choose from with links to CC licence deeds. It might offer  a link to a "help guide" on how to determine which license is correct.  If the web page is not from a whitelist of good sites (e.g Wikimedia Commons or other site with open licenses) it may offer a warning or disclaimer. It may offer some non-CC license option(s), albeit with a warning about reuse; since even an image with a restrictive copyright can, under certain conditions, be "fair use" for a scholar to acquire for personal use.    

This could have editable fields, for the title, URL, person/maker, organization, etc. in case these were not quite right or missing. It might use these fields and license to suggest an attribution string in plain text, markdown, HTML--or dare one hope, eventually, LaTex format to add to filename.txt. 

One could see this function as a companion extension to this one or even be an option in current extension--e.g an optional choice: do you wish to acquire an image and corresponding attribution from a search or from this page.  Seems a political choice that 
Creative Commons would need to make if such a feature is "In Scope" and a priority. I could understand if they feel not, but I wish someone could build this!

Thanks again!

**Comment by kgodey** [comment]
@jaysnoddy Thank you for the suggestion! The extension that you are describing sounds very useful, but I'm not sure that it would be a good additional feature for this particular extension, since this is just designed as an interface for the existing CC Search functionality. A companion extension would make sense, but I don't think CC has the resources to work on it any time soon.

I will file the idea for a future internship program or to reevaluate when we have more resources available.
